### PR TITLE
pscanrulesAlpha: Added Modern App Detection Scanner

### PR DESCRIPTION
**Pull request by psiinon**
The rule id is added in https://github.com/zaproxy/zaproxy/pull/5446

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    private PassiveScanThread parent = null;
+
+    /** Prefix for internationalized messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanalpha.modernapp.";
+
+    /**
+     * gets the name of the scanner
+     *
+     * @return
```
Comment (kingthorin) [inline addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java]: Remove all the boilerplate javadoc or fill it out? Throughout the class

File: addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        if (!msg.getResponseHeader().isHtml()) {
+            // Only check HTML responses
+        }
+        String evidence = null;
+        String otherInfo = null;
+
+        List<Element> links = source.getAllElements(HTMLElementName.A);
+        if (links.size() == 0) {
```
Comment (kingthorin) [inline addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java]: Could this be null? (I don’t recall the behaviour of getAllElements (Same for the one below)

File: addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java (hunk 1 above)
Comment (psiinon) [reply addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java]: yeah, just ripped off an existing one ;)

File: addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java (hunk 2 above)
Comment (psiinon) [reply addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java]: No it cant - the tests confirm this :D

File: addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java (hunk 1 above)
Comment (psiinon) [reply addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/ModernAppDetectionScanner.java]: Removed

**Pull request merged**